modAionImpl/src/org/aion/zero/impl/cli/Cli.java

```diff
@@ -221,7 +223,75 @@ public ReturnType call(final String[] args, Cfg cfg) {
                 // no return, allow for other parameters combined with -p
             }
 
-            // 4. can be influenced by the -n, -d, -p arguments above
+            // determine the sync compact configuration, can be combined with the -n, -d, -c, -p, -i
+            // arguments
+
+            if (options.getCompact() != null) {
+
+                CfgSync cfgSync = cfg.getSync();
+                String[] parameters = options.getCompact();
+
+                if (parameters.length == 1) {
+
+                    if (!parameters[0].equalsIgnoreCase("true")
+                            && !parameters[0].equalsIgnoreCase("false")) {
+                        System.out.println("enabled value must be boolean");
+                    }
+                    boolean compactEnabled = Boolean.parseBoolean(parameters[0]);
+                    if (compactEnabled != cfgSync.getCompactEnabled()) {
+                        cfgSync.setCompactEnabled(compactEnabled);
+                        overwrite = true;
+                    }
+                    System.out.println("Compact enabled is set to: " + compactEnabled);
+
+                } else if (options.getCompact().length == 2) {
+
+                    int slowImportTime = cfgSync.getSlowImportTime();
+                    int compactFrequency = cfgSync.getCompactFrequency();
+                    boolean validCompact = true;
+
+                    try {
+                        slowImportTime = Integer.parseInt(parameters[0]);
+                        compactFrequency = Integer.parseInt(parameters[1]);
+                    } catch (NumberFormatException e) {
+                        validCompact = false;
+                        System.out.println(
+                                "slow_import and frequency values must be positive integers");
+                    }
+                    if (slowImportTime < 0 || compactFrequency < 0) {
```

Comment:
Do we want to allow 0? Should it be `if (slowImportTime <= 0 || compactFrequency <= 0)`?